services/motion/builtin/move_request.go

```diff
@@ -175,70 +255,47 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 		return state.ExecuteResponse{}, err
 	}
 
-	for visSrvc, cameraNames := range mr.obstacleDetectors {
-		for _, camName := range cameraNames {
-			mr.logger.Debugf(
-				"proceeding to get detections from vision service: %s with camera: %s",
-				visSrvc.Name().ShortName(),
-				camName.ShortName(),
-			)
+	// if the camera is mounted on something InputEnabled that isn't the base, then that
+	// input needs to be known in order to properly calculate the pose of the obstacle
+	// furthermore, if that InputEnabled thing has moved since this moveRequest was initialized
+	// (due to some other non-motion call for example), then we can't just get current inputs
+	// we need the original input to place that thing in its original position
+	// hence, cached CurrentInputs from the start are used i.e. mr.planRequest.StartConfiguration
+	existingGifs, err := mr.planRequest.WorldState.ObstaclesInWorldFrame(
+		mr.planRequest.FrameSystem, mr.planRequest.StartConfiguration,
+	)
+	if err != nil {
+		return state.ExecuteResponse{}, err
+	}
+	// existingGifs are in their relative position, i.e. with respect to the base frame
+	// here we transform them into their absolute positions. i.e. with respect to the world frame
+	existingGeoms := []spatialmath.Geometry{}
+	for _, g := range existingGifs.Geometries() {
+		existingGeoms = append(existingGeoms, g.Transform(mr.poseOrigin))
+	}
+	absoluteExistingGifs, err := referenceframe.NewGeometriesInFrame(referenceframe.World, existingGeoms), nil
```

Comment:
I don't think it makes sense to have static geometries be considered here since they would be re-detected by the vision service. This would also allow us to simplify this section. @raybjork @biotinker

Reply:
I recommend against this change.

There is no requirement that *any* vision service be configured, or that it can definitely detect the obstacle in question, or anything else.

Ultrasonic sensors for example have very low resolution.

Reply:
ah, good argument. I will add back.